classes/class-swsales-reports.php

```diff
@@ -42,141 +73,263 @@ public static function show_reports_page() { ?>
 				);
 
 				// Choose sale to show.
-				$sale_to_show = null;
+				$sales_to_show = array();
 				if ( isset( $_REQUEST['sitewide_sale'] ) ) {
-					$sale_to_show = SWSales_Sitewide_Sale::get_sitewide_sale( $_REQUEST['sitewide_sale'] );
+					if ( ! is_array( $_REQUEST['sitewide_sale'] ) ) {
+						$_REQUEST['sitewide_sale'] = array( $_REQUEST['sitewide_sale'] );
+					}
+
+					foreach ($_REQUEST['sitewide_sale'] as $key => $value) {
+						if ( ! empty( $value ) ) {
+							$sales_to_show[] = SWSales_Sitewide_Sale::get_sitewide_sale( (int)$value );
+						}
+					}
 				}
-				if ( null === $sale_to_show ) {
-					$sale_to_show = SWSales_Sitewide_Sale::get_active_sitewide_sale();
+				if ( empty( $sales_to_show ) && ! empty( SWSales_Sitewide_Sale::get_active_sitewide_sale() ) ) {
+					$sales_to_show[] = SWSales_Sitewide_Sale::get_active_sitewide_sale();
 				}
 
-				// Select field to choose a sitewide sale.
+				// Show dropdown to choose sale to show.
 				if ( ! empty ( $all_sitewide_sales ) ) { ?>
 					<form method="get" action="/wp-admin/edit.php">
 						<input type="hidden" name="post_type" value="sitewide_sale" />
 						<input type="hidden" name="page" value="sitewide_sales_reports" />
-						<label for="sitewide_sale"><?php esc_html_e( 'Show reports for', 'sitewide-sales' ); ?></label>
-						<select id="swsales_select_report" name="sitewide_sale" onchange="this.form.submit()">
-							<?php
-							foreach ( $all_sitewide_sales as $sitewide_sale_id ) {
-									$sale              = SWSales_Sitewide_Sale::get_sitewide_sale( $sitewide_sale_id );
-									$selected_modifier = ( ! ( null === $sale_to_show ) && $sale->get_id() === $sale_to_show->get_id() ) ? 'selected="selected"' : '';
-								?>
-								<option value="<?php esc_attr_e( $sale->get_id() ); ?>" <?php echo( esc_html( $selected_modifier ) ); ?>>
-									<?php echo( esc_html( $sale->get_name() ) ); ?>
-								</option>
+						<div class="swsales_reports-filters">
+							<?php 
+
+							for ($i = 0; $i < 2; $i++) {
+								$class_modifier = ( $i === 0 ) ? 'left' : 'right';
+								$label_wording = ( $i === 0 ) ? 'Show reports for' : 'vs.';
+								
+								$selected =  $sales_to_show !== null && array_key_exists( $i, $sales_to_show )  ? $sales_to_show[$i] : null;
+							?>
+								<label for="swsales_select_report_<?php echo $class_modifier ?>"><?php esc_html_e( $label_wording, 'sitewide-sales' ); ?></label>
+								<select id="swsales_select_report_<?php echo $class_modifier ?>"  name="sitewide_sale[]">
+									<?php
+									if($i == 1) {
+									?>
+									<option selected value="0"><?php esc_html_e( '- Choose One -', 'sitewide-sales' ); ?></option>
+									<?php
+									}
+									foreach ( $all_sitewide_sales as $sitewide_sale_id ) {
+										$sale   = SWSales_Sitewide_Sale::get_sitewide_sale( $sitewide_sale_id );
+										$selected_modifier =  $selected != null && $sale === $selected ? 'selected="selected"' : '';
```

Comment:
Probably not worth changing, but could consider making sure $selected is never null (initialize as '') and using the WP function inline in the option tag below.

https://developer.wordpress.org/reference/functions/selected/

Reply:
I worked on this a bit but its not liking anyting I try. Not changing for now.